Source/Utils/WarpXUtil.cpp

```diff
@@ -398,6 +399,28 @@ void ReadBCParams ()
                 }
             }
         }
+#ifdef WARPX_USE_PSATD
+        int maxwell_solver_id = GetAlgorithmInteger(pp_algo, "maxwell_solver");
+        if (maxwell_solver_id == MaxwellSolverAlgo::PSATD) {
+            ParmParse pp_psatd("psatd");
+            int do_moving_window_input = 0;
+            pp_warpx.query("do_moving_window", do_moving_window_input);
+            if (do_moving_window_input == 1) {
+                std::string s;
+                pp_warpx.get("moving_window_dir", s);
+                int zdir;
+                if (s == "z" || s == "Z") zdir = AMREX_SPACEDIM-1;
+                bool damp_fields_in_z_guard = true;
+                if (pp_psatd.query("use_damp_fields_in_z_guard", damp_fields_in_z_guard) ) {
+                    amrex::Warning("WARNING : the input parameter ``use_damp_fields_in_z_guard`` will be deprecated soon. Please use boundary.field_lo=Damped and boundary.field_hi=Damped instead.");
+                }
+                if (damp_fields_in_z_guard == true) {
+                    WarpX::field_boundary_lo[zdir] = FieldBoundaryType::Damped;
+                    WarpX::field_boundary_hi[zdir] = FieldBoundaryType::Damped;
+                }
```

Comment:
Just a question:
Would it be possible to already remove the old syntax `use_damp_fields_in_z_guard`, instead of keeping both of them for a while?

If so, we could simply remove it and abort the code (instead of issuing a warning) if a user sets the old parameter, setting the call to `amrex::Abort` in the function `WarpX::BackwardCompatibility` in Source/WarpX.cpp. This would mean maybe removing the whole code block here and having the check/abort in `WarpX::BackwardCompatibility`.

Reply:
See my comment elsewhere - I agree that the `use_damp_fields_in_z_guard` can be removed, though I don't think it is necessary to include it in `BackwardCompatibility`. This block of code is not needed. It is Ok to require use of the new boundary condition input parameters to turn this feature on since it is a new feature.